src/components/space/charts/MemberActionsChart.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+
+import { Typography } from '@mui/material';
+
+import { Action } from '@graasp/sdk';
+
+import { endOfWeek, format } from 'date-fns';
+import { groupBy } from 'lodash';
+import {
+  Bar,
+  CartesianGrid,
+  ComposedChart,
+  Legend,
+  Tooltip,
+  XAxis,
+  YAxis,
+} from 'recharts';
+
+import ChartContainer from '@/components/common/ChartContainer';
+import { getColorForActionTriggerType } from '@/config/constants';
+import { useActionsTranslation, useAnalyticsTranslation } from '@/config/i18n';
+
+type Props = {
+  actionsGroupedByWeekStart: { [key: string]: Action[] };
+};
+const MemberActionsChart = ({
+  actionsGroupedByWeekStart,
```

Comment:
I noticed that if I choose only one week, I can only see one bar. It would be better in that for less than a week, the actions are displayed per day. It means that it would be better for the chart to handle how it groups the actions, rather than receiving them already grouped.
Maybe this function can be useful to automatically decide what range is the best to group by https://date-fns.org/v3.6.0/docs/intervalToDuration

Reply:
There's also a problem if I select a very long period of time
<img width="1136" alt="Screenshot 2024-06-07 at 10 48 59" src="https://github.com/graasp/graasp-analytics/assets/11229627/98aa6233-1b9b-4b2b-9463-577f61786690">